opm/output/Cells.hpp

```diff
@@ -0,0 +1,84 @@
+/*
+  Copyright 2016 Statoil ASA.
+
+  This file is part of the Open Porous Media project (OPM).
+
+  OPM is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  OPM is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef OPM_OUTPUT_CELLS_HPP
+#define OPM_OUTPUT_CELLS_HPP
+
+#include <map>
+#include <vector>
+
+namespace Opm {
+
+class SimulationDataContainer;
+
+namespace data {
+
+    struct Solution {
+        /* data::Solution supports writing only some information,
+         * distinguished by keys. When adding support for more values in
+         * the future, add a proper key.
+         */
+        enum class key {
+            PRESSURE,
+            TEMP,
+            SWAT,
+            SGAS,
+            RS,
+            RV,
+        };
+
+        inline bool has( key ) const;
+
+        inline const std::vector< double >& operator[]( key ) const;
+        inline std::vector< double >& operator[]( key );
+
+        void insert( key, std::vector< double > );
```

Comment:
For sanity checks it would make sense to check the size of the input vector; the expected number of cells could be a member variable specified in the constructor.